udf-api/src/main/java/org/apache/iotdb/udf/api/type/Binary.java

```diff
@@ -85,7 +89,10 @@ public boolean equals(Object other) {
 
   @Override
   public int hashCode() {
-    return Arrays.hashCode(values);
+    if (hash == 0) {
+      hash = Arrays.hashCode(values);
```

Comment:
there's a corner case... If the result just equals 0, then it will be computed every time... Maybe we should add another boolean value to indicate whether the hashCode has been calculated...

Reply:
Right... Using Integer needs Boxing and Unboxing, I added a boolean value.